docs/specs/online_store_format.md

```diff
@@ -59,6 +59,34 @@ Here's an example of how the entire thing looks like:
 
 However, we'll address this issue in future versions of the protocol.
 
+## Cloud Firestore Online Store Format
+
+[Firebase data model](https://firebase.google.com/docs/firestore/data-model) is a hierarchy of documents that can contain (sub)-collections. This structure can be multiple levels deep; documents and subcollections are alternating in this hierarchy.
+
+We use the following structure to store feature data in the Firestore:
+* at the first level, there is a collection for each Feast project
+* second level, in each project-collection, there is a Firebase document for each Feature Table
+* third level, in the document for the Feature Table, there is a subcollection called `values` that contain a document per feature row. That document contains the following fields:
+  * `key` contains entity key as serialized `feast.types.EntityKey` proto
+  * `value` contains value as serialized `feast.types.Value` proto
+  * `event_ts` contains event timestamp (in the native firestore timestamp format)
+  * `created_ts` contains write timestamp (in the native firestore timestamp format)
+
+Document id for the feature document is computed by hashing entity key using murmurhash3_128 algorithm as follows:
```

Comment:
How is the Document Id used?

Reply:
Document ID is basically the primary key, so on read path you'd compute it and use to retrieve the doc containing feature values.

Another thing I've considered here is to use key field in the doc itself, since Firestore automatically creates indices it would also be pretty efficient. However, in that case it would be theoretically possible to have two docs containing the same entity key, and I wanted to avoid thinking through an additional failure mode.

Reply:
Makes sense.